common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
```

Comment:
I dislike this API because I do not feel it is clear what it does and it creates ambiguity in regards to what `authorise` does.  For example, if you are using a REST library and call `authorise({ force: true })`, it will obtain a new token (if a means to generate one is available) and use that for all new requests.  However, in the case of realtime, it will not until the connection is re-established.  Then calling `setOptions` in the case of REST will do the same as `authorise({ force: true})`

I therefore propose we instead simply change the specification stating that calling `authorise({ force: true })` will action the current connection to be upgraded in the case of connected Realtime client libraries. I can't think of many use cases where the user would not want that to happen.

The major issue I foresee with either approach at this stage is that if a user attempts to reauthorise and the new token details are incompatible with the old, then instead of simply rejecting the upgrade attempt, the connection will effectively move to the `FAILED` state and all state will be lost.  I feel this is perhaps something we can live with until we support upgrade authentication in the protocol itself.

Also, @paddybyers can you confirm what happens if the user is assigned less privileges than before, will that be rejected or will the new privileges be applied and if any channels are attached that are not supported, they will be detached. I assume there are cases where privileges may be downgraded and I think we should support probably this.